File: leet_code/javascript_ques/go/2620.go
package main

import "fmt"

func createCounter(n int) func() int {
	counter := n - 1
	return func() int {
		counter++
		return counter
	}
}

func createCounter2(n int) func() int {
	counter := n
	return func() int {
		counter++
		return counter - 1
	}
}

func main() {
	counter := createCounter(5)
	fmt.Println(counter()) // 5
	fmt.Println(counter()) // 6

	counter2 := createCounter2(5)
	fmt.Println(counter2()) // 5
	fmt.Println(counter2()) // 6
}

File: leet_code/javascript_ques/python/2620.py
def create_counter(n):
    counter = n - 1
    def counter_function():
        nonlocal counter
        counter += 1
        return counter
    return counter_function


def create_counter2(n):
    counter = n - 1
    def counter_function():
        nonlocal counter
        counter += 1
        return counter
    return counter_function
